text/00110-cli-framework-compatibility-strategy.md

```diff
@@ -0,0 +1,391 @@
+    - `pass-on-error`: Tests are executed with expectation to fail with an error message. And indeed the error message is correct.    
+    - `fail`: Tests are executed with an expectation to pass, but some of them don't.
+    - `fail-on-error`: Tests are executed with an expectation to fail with an error message, but the error message is incorrect.
+    
+4. Depending of which state, the developer performs one of the following actions:
+
+    - `make-compatible`: Try to make the code compatible. This would normally happen when the breakage is unintentional, and therefore no version bump was performed.
+    - `mark-incompatible`: Developer reluctantly understands that the changes cannot be compatible, and decides to bump the version. In addition, the test is amended to mark this version as incompatible.
+    - `fix-version-bump`: This would happen when a version bump was performed, but the value it was bumped to is incorrect.
+
+5. If the state of step 3 is `pass` or `pass-on-error`, we are done. Otherwise, *GOTO* 1.
+
+<br>
+
+<!-- To edit: URL Decode -> Edit -> URL Encode -->
+
+<img src='https://g.gravizo.com/svg?digraph%20G%20%7B%0A%20%20%20%20aize%20%3D%224%2C4%22%3B%0A%20%20%20%20%22merge-to-master%22%3B%0A%20%20%20%20%22merge-to-master%22%20-%3E%20%22test-framework-compatibility.sh%22%3B%0A%20%20%20%20node%20%5Bshape%3Dbox%2Cstyle%3Dfilled%2Ccolor%3D%22yellowgreen%22%5D%3B%0A%20%20%20%20%22test-framework-compatibility.sh%22%20-%3E%20pass%3B%0A%20%20%20%20%22test-framework-compatibility.sh%22%20-%3E%20%22pass-on-error%22%3B%0A%20%20%20%20node%20%5Bshape%3Dbox%2Cstyle%3Dfilled%2Ccolor%3D%22crimson%22%5D%3B%0A%20%20%20%20%22test-framework-compatibility.sh%22%20-%3E%20%22fail%22%3B%0A%20%20%20%20%22test-framework-compatibility.sh%22%20-%3E%20%22fail-on-error%22%3B%0A%20%20%20%20%22fail%22%20-%3E%20%22merge-to-master%22%20%5Blabel%20%3D%20%22%20%20make-compatible%5CnOR%5Cnmark-incompatible%22%5D%0A%20%20%20%20%22fail-on-error%22%20-%3E%20%22merge-to-master%22%20%5Blabel%20%3D%20%22%20%20fix-version-bump%22%5D%0A%20%20%7D%0A'/>
+
+<br>
+
+# Drawbacks
+
+- Dramatically increases the duration of our integration tests suite.
+- Requires manually editing the framework compatibility test file.
+
+    > This isn't necessarily a drawback, as it provides a clear paper trail of all our breakages.
+
+# Rationale and Alternatives
+
+An alternative approach would to be force CLI versions to *always* be `>=` than Framework Versions. This differs from our current statement where we do try to have newer frameworks support older CLI as much as possible.
+
+Doing so drastically simplify the entire process, as it means we can effectively drop the entire notion of *cx-protocol*.
+
+## Why is it OK to enforce this requirement?
+
+- We want to migrate most of the functionality to the framework. Making CLI upgrades especially safe.
```

Comment:
Making it also pointless to require upgrading the CLI if nothing effectively changed in the CLI code.

Reply:
You are correct